source/dpp/translation/aggregate.d

```diff
@@ -512,7 +512,8 @@ private string[] maybeC11AnonymousRecords(in from!"clang".Cursor cursor,
     static bool isFieldOfRightType(in Cursor member, in Cursor child) {
         const isField =
             child.kind == Cursor.Kind.FieldDecl &&
-            child.type.canonical == member.type.canonical;
+            (child.type.canonical == member.type.canonical ||
```

Comment:
At this point, child.type.canonical would be:
Type(Elaborated, "const struct (anonymous struct at ./ceva.h:22:12)")

and member.type.canonical:                            
Type(Record, "struct A::(anonymous at ./ceva.h:22:12)")

That is why I chose to check against the unelaborated type.

Reply:
The comment you left on the diff should be a comment in the code.